### PR TITLE
cql3,db: call storage_options::from_map using opts.from_map()

### PR DESCRIPTION
**Pull request by tchaikov**
simpler this way. and for better readability.

### CODE REVIEW

File: cql3/statements/ks_prop_defs.cc [hunk 1]
```diff
@@ -101,7 +101,7 @@ data_dictionary::storage_options ks_prop_defs::get_storage_options() const {
         if (it != options_map->end()) {
             sstring storage_type = it->second;
             options_map->erase(it);
-            opts.value = data_dictionary::storage_options::from_map(storage_type, std::move(*options_map));
+            opts.value = opts.from_map(storage_type, std::move(*options_map));
```
Comment (nyh) [inline cql3/statements/ks_prop_defs.cc:104]: Why is this simpler or more readable?
The from_map() function is **static**, so traditionally the class name, not an object, is used to call it. Especially in this case, where the whole idea of from_map is to create a new object, unrelated to the old object, so the "opts.from_map()" way of calling it obscures this - it makes the reader think maybe the result somehow depends on the old "opts", which it doesn't.

**Review by nyh — COMMENTED** [review]
I don't think this is an improvement (see inline), but it's an issue of style, so feel free to ask other people, and they might favor this change.

File: cql3/statements/ks_prop_defs.cc (hunk 1 above)
Comment (tchaikov) [reply cql3/statements/ks_prop_defs.cc:104]: simpler in terms of cognitive load put on programmer that he/she needs to connect `data_dictionary::storage_options` to opts. 

> Especially in this case, where the whole idea of from_map is to create a new object, unrelated to the old object, so the "opts.from_map()" way of calling it obscures this - it makes the reader think maybe the result somehow depends on the old "opts", which it doesn't.

okay. that's another way to look at this code. and it makes sense also. i am closing this PR.

**Comment by tchaikov** [comment]
> I don't think this is an improvement (see inline), but it's an issue of style, so feel free to ask other people, and they might favor this change.

thank you for your input. i understand this is in a gray area. will wait for a while, if no more inputs by then, i will close this PR.

**Pull request closed without merging**